### PR TITLE
Scrape github for time of last contribution to zulip.

### PR DESCRIPTION
**Pull request by cosmicAsymmetry**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@cosmicAsymmetry, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by cosmicAsymmetry** [comment]
I wasn't able to scrape the text like, "last updated 6 hours ago" as that it processed later by the JS. I could spend a little more time to implement that directly in py if you like.

**Comment by umairwaheed** [comment]
@cosmicAsymmetry, you will have to make this an executable script. You also need to add `argparse` module to handle arguments supplied by the user.

File: scripts/github-last-contribution.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+'''
+
+
+def last_contribution(user_name):
+    profile_url = "https://github.com/{}/zulip".format(user_name)
+    try:
+        profile_page = BeautifulSoup(
+            urllib.request.urlopen(profile_url), "lxml")
+    except urllib.error.URLError:
+        return "Error"
```
Comment (TomaszKolek) [inline scripts/github-last-contribution.py]: I'd like to see a reason too (e.g. service unreachable)

File: scripts/github-last-contribution.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    name_and_time_stamp = [user_name] + time_stamp
+    return name_and_time_stamp
+
+
+def get_usernames(user_names):
+    user_names = user_names.split(' ')
+    print("Here are the timestamps for the last commit to the Zulip forks:")
+    for user_name in user_names:
+        name_and_time_stamp = last_contribution(user_name)
+        if name_and_time_stamp == "Error":
```
Comment (TomaszKolek) [inline scripts/github-last-contribution.py]: I think its better to raise an error in above function and just catch it here. Returning 'Eror' does not look like good practice

File: scripts/github-last-contribution.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            print("{} is an invalid user or has not forked the Zulip repository.".format(
+                user_name))
+        else:
+            print("{} last updated on {}, {}".format(*name_and_time_stamp))
+
+
+def Main():
```
Comment (TomaszKolek) [inline scripts/github-last-contribution.py]: function should be always written lowercase.

Comment (cosmicAsymmetry) [inline scripts/github-last-contribution.py]: Fixing ASAP.

**Comment by cosmicAsymmetry** [comment]
Hey @TomaszKolek comments addressed!

File: scripts/github-last-contribution.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def last_contribution(user_name):
+    profile_url = "https://github.com/{}/zulip".format(user_name)
+    try:
+        profile_page = BeautifulSoup(
+            urllib.request.urlopen(profile_url), "lxml")
+    #404 page not reachable or page doesn't exist.
+    except urllib.error.URLError:
+        return urllib.error.URLError
```
Comment (cosmicAsymmetry) [inline scripts/github-last-contribution.py]: What error should be raised? The one returned from urllib, right?

Comment (cosmicAsymmetry) [inline scripts/github-last-contribution.py]: Apologies for bothering you so much. This is my first time using except and raise.

File: scripts/github-last-contribution.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def get_usernames(user_names):
+    user_names = user_names.split(' ')
+    print("Here are the timestamps for the last commit to the Zulip forks:")
+    for user_name in user_names:
+        try:
+            name_and_time_stamp = last_contribution(user_name)
+            print("{} last updated on {}, {}".format(*name_and_time_stamp))
+        except:
```
Comment (TomaszKolek) [inline scripts/github-last-contribution.py]: Here you should be more specific (you want to catch just particular kind of errors)

`except MyError:`

Comment (cosmicAsymmetry) [inline scripts/github-last-contribution.py]: Okay.

File: scripts/github-last-contribution.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    time_stamp = time_tag[0].get("datetime")[:-1].split("T")
+    name_and_time_stamp = [user_name] + time_stamp
+    return name_and_time_stamp
+
+
+def get_usernames(user_names):
+    user_names = user_names.split(' ')
+    print("Here are the timestamps for the last commit to the Zulip forks:")
+    for user_name in user_names:
+        user_name = lower(user_name)
```
Comment (TomaszKolek) [inline scripts/github-last-contribution.py]: Did you run it? Lower is not defined

**Comment by TomaszKolek** [comment]
```
Traceback (most recent call last):
  File "to_remove.py", line 50, in <module>
    main()
  File "to_remove.py", line 47, in main
    get_usernames(" ".join(args.usernames))
  File "to_remove.py", line 36, in get_usernames
    name_and_time_stamp = last_contribution(user_name)
  File "to_remove.py", line 24, in last_contribution
    time_stamp = time_tag[0].get("datetime")[:-1].split("T")
IndexError: list index out of range
```
`python3 script.py TomaszKolek`

I'd like to encourage you to test it well before sending to review.

File: scripts/github-last-contribution.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+import argparse
+import urllib.request
+from bs4 import BeautifulSoup
+'''
```
Comment (umairwaheed) [inline scripts/github-last-contribution.py]: This comment should probably come right after the shebang (#!/usr/bin/env python).

**Comment by umairwaheed** [comment]
@cosmicAsymmetry, looks good and works. Can you use underscores in the name of your module instead of dashes? This way your module will be importable.

**Comment by cosmicAsymmetry** [comment]
Hey @umairwaheed, thank you so much for going through my code. Isn't all my code already in underscores? Or are you referring to the file name?

**Comment by hackerkid** [comment]
❯ python3 last_contribution.py hackerkid            
```py
Here are the timestamps for the last commit to the Zulip forks:
Traceback (most recent call last):
  File "test.py", line 46, in <module>
    main()
  File "test.py", line 43, in main
    get_usernames(" ".join(args.usernames))
  File "test.py", line 31, in get_usernames
    name_and_time_stamp = last_contribution(user_name)
  File "test.py", line 21, in last_contribution
    time_stamp = time_tag[0].get("datetime")[:-1].split("T")
IndexError: list index out of range
```

**Comment by umairwaheed** [comment]
>Hey @umairwaheed, thank you so much for going through my code. Isn't all my code already in underscores? Or are you referring to the file name?

Yep, the filenames.

**Pull request closed without merging**